src/features/transfer/TransferTokenForm.tsx

```diff
@@ -335,21 +365,43 @@ function validateFormValues(
   { originChainId, destinationChainId, amount, tokenAddress, recipientAddress }: TransferFormValues,
   queryClient: QueryClient,
   accountAddress?: string,
+  isERC721?: boolean,
 ) {
   if (!originChainId) return { originChainId: 'Invalid origin chain' };
   if (!destinationChainId) return { destinationChainId: 'Invalid destination chain' };
   if (!isValidAddress(recipientAddress)) return { recipientAddress: 'Invalid recipient' };
   if (!isValidAddress(tokenAddress)) return { tokenAddress: 'Invalid token' };
   const parsedAmount = tryParseAmount(amount);
-  if (!parsedAmount || parsedAmount.lte(0)) return { amount: 'Invalid amount' };
+
+  if (!parsedAmount || parsedAmount.lte(0))
+    return isERC721 ? { amount: 'Invalid Token Id' } : { amount: 'Invalid amount' };
+  // Valicate erc721
+  // use when user input token id
+  const cachedOwnerOf = getCachedOwnerOf(queryClient, originChainId, tokenAddress, amount);
+  // use when user select token id
+  const cachedTokenIdBalance = getCachedTokenIdBalance(
+    queryClient,
+    originChainId,
+    tokenAddress,
+    accountAddress,
+  );
+  if (
+    isERC721 &&
+    ((cachedOwnerOf && accountAddress !== cachedOwnerOf) ||
+      !cachedTokenIdBalance?.includes(amount.toString()))
```

Comment:
How does this work for tokens that don't support `tokenIdByOwnerIndex`?

Reply:
- for token tokens which support tokenOfOwnerByIndex 
 
i have a hook that get all tokenIds and show as a select component then i check that list of token have contains or not
```
!cachedTokenIdBalance?.includes(amount.toString())
```
- for token tokens which don't support tokenOfOwnerByIndex 

i show input component and when user input something it will call `useOwnerOfErc721()` then we have `cachedOwnerOf` value to validate